server/src/instant/db/transaction.clj

```diff
@@ -226,36 +226,70 @@
+                                                                   [:= :entids_forward.etype :refs_forward.reverse_etype]]]
+                                           :select [:refs_forward.entity_id :refs_forward.forward_etype]}]}]
+                                
+                                [[:entids_reverse {:columns [:entity_id :etype]}]
+                                 {:union [{:from :entids_forward
+                                           :select [:entity_id :etype]}
+                                          {:from   :entids_reverse
+                                           :join   [:refs_reverse [:and
+                                                                   [:= :entids_reverse.entity_id :refs_reverse.entity_id]
+                                                                   [:= :entids_reverse.etype :refs_reverse.forward_etype]]]
+                                           ;; TODO value -> value_ref
+                                           :select [[[:cast [:->> :refs_reverse.value :0] :uuid]] :refs_reverse.reverse_etype]}]}]
+                                
+                                [[:attrs_forward {:columns [:id :forward_etype :reverse_etype]}]
+                                 (if (empty? attrs+etypes)
+                                   {:select [[[:cast nil :uuid]] nil nil] :where false}
+                                   {:values attrs+etypes})]
+                                
+                                [:refs_forward {:select [:triples.entity_id :triples.value :attrs_forward.forward_etype :attrs_forward.reverse_etype]
+                                                :from   :triples
+                                                :join   [:attrs_forward [:= :triples.attr_id :attrs_forward.id]]
+                                                :where  [:and
+                                                         :triples.vae
+                                                         [:= :triples.app_id app-id]]}]
+                                
+                                [[:attrs_reverse {:columns [:id :forward_etype :reverse_etype]}]
+                                 (if (empty? reverse-attrs+etypes)
+                                   {:select [[[:cast nil :uuid]] nil nil] :where false}
+                                   {:values reverse-attrs+etypes})]
+                                
+                                [:refs_reverse {:select [:triples.entity_id :triples.value :attrs_reverse.forward_etype :attrs_reverse.reverse_etype]
+                                                :from   :triples
+                                                :join   [:attrs_reverse [:= :triples.attr_id :attrs_reverse.id]]
+                                                :where  [:and
+                                                         :triples.vae
+                                                         [:= :triples.app_id app-id]]}]]
+               :from   :entids_reverse
```

Comment:
Would it be better to do:

```clojure
:union [{:select :* :from :entids_reverse}
        {:select :* :from :entids_forward}]
```

here instead of doing the union in the `entids_reverse` cte?

Reply:
Solved